### PR TITLE
Change installation

### PR DESCRIPTION
**Pull request by JamesB-1qbit**
Remove pyscf dependency.

### CODE REVIEW

**Comment by ValentinS4t1qbit** [comment]
I see the main branch test has failed: it's possible that you fixed the target branch afterwards, but Github will only notice and rerun it properly after you push something new. Let's merge the other PRs first, this way you'll update your branches gradually, push, and will naturally fix that.

In the meantime, I'll look into the rest

File: README.rst [hunk 1]
```diff
@@ -124,16 +124,20 @@ Depending on your OS and environment, some of these packages may be more challen
...
 
-Although PySCF is installed by default, this behaviour can be turned off by defining the environment variable `NO_PYSCF=1` and then
-installing using pip by `pip install git+https://github.com/goodchemistryco/Tangelo.git@main` or using the "From source, using setuptools"
-method. 
+The two quantum chemistry packages that are natively supported are `PySCF <https://pyscf.org/>`_ and `Psi4 <https://psicode.org/>`_.
 
-The classical solvers `CCSDSolver`, `MP2Solver`, `FCISolver` and `MINDO3Solver` as well as `DMETProblemDecomposition` require 
-PySCF at this time.
+`DMETProblemDecomposition` is only supported with `PySCF`.
+`SemiEmpircalSolver` requires PySCF  along with 'python3 -m pip install git+https://github.com/pyscf/semiempirical'.
```
Comment (ValentinS4t1qbit) [inline README.rst:135]: I think we can delete that too. It's easy to figure out, I believe we handle it with ModuleNotFound, and it's only relevant for people who actually call the Semi-Empirical solver and... to my knowledge that's no one so far. Let's not get our already-crowded README more cluttered with minor things.

File: README.rst [hunk 2]
```diff
@@ -124,16 +124,20 @@ Depending on your OS and environment, some of these packages may be more challen
...
 
-Psi4 is supported everywhere except `CCSDSolver`, `MP2Solver`, `FCISolver` and `MINDO3Solver` and `DMETProblemDecomposition`.
+You are also welcome to provide your own interface to a quantum chemistry package of your choice by defining a subclass of
```
Comment (ValentinS4t1qbit) [inline README.rst:137]: We have a test that illustrates that: instead of trying to explain it, a link to the test I think would give much more. Let's be as concise as possible and reuse our resources rather than "duplicate" or "reformulate" what already exists.

A good question is whether or not we should inject / illustrate the feature in a notebook or not (maybe not worth the hassle). We don't have to mention a notebook or get a notebook ready for this PR. Let's just think about whether it's desirable, write it down here, and resolve this comment.

File: README.rst [hunk 3]
```diff
@@ -124,16 +124,20 @@ Depending on your OS and environment, some of these packages may be more challen
 or any issue regarding the above packages, please check their respective documentation.
 
 
-Having trouble with PySCF?
+Quantum Chemistry Packages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Tangelo can be used without having a quantum chemistry package installed but many algorithms, by default, depend on one being installed.
```
Comment (ValentinS4t1qbit) [inline README.rst:130]: I think we need to specify "classical" here, and say what are the main things these package does (compute electronic integrals ? Provide reference classical solvers for HF and post-HF techniques ?)

@AlexandreF-1qbit how would you concisely answer this question ?

File: README.rst [hunk 4]
```diff
@@ -124,16 +124,20 @@ Depending on your OS and environment, some of these packages may be more challen
 or any issue regarding the above packages, please check their respective documentation.
 
 
-Having trouble with PySCF?
+Quantum Chemistry Packages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (ValentinS4t1qbit) [inline README.rst:128]: too many ^^

File: .github/workflows/run_psi4_test.yml [hunk 5]
```diff
@@ -2,13 +2,10 @@ name: psi4 test
...
 on: [pull_request]
 
-env:
-  NO_PYSCF: 1
-
 jobs:
   build:
 
-    runs-on: ubuntu-latest
+    runs-on: windows-latest
```
Comment (ValentinS4t1qbit) [inline .github/workflows/run_psi4_test.yml:8]: I like that we are putting windows into the picture, but now we're a bit all over the place.

- why this change ?
- does that mean we have tests running on ubuntu with pypi , but windows with psi4 ? Do we believe that's what our users are mostly using ?

As a result of this change, are we no longer testing a configuration that we should be testing ?

File: README.rst [hunk 6]
```diff
@@ -124,16 +124,20 @@ Depending on your OS and environment, some of these packages may be more challen
 or any issue regarding the above packages, please check their respective documentation.
 
 
-Having trouble with PySCF?
+Quantum Chemistry Packages
```
Comment (ValentinS4t1qbit) [inline README.rst:127]: We had another section just before that saying "optional dependencies".
Can we make things flow a bit better ?

It feels like we now have a optional dependency section with a first "sub section" about quantum circuit backends, and another one about classical quantum chemistry backends. Maybe those are the "sub titles" we need, we throw 1-2 links and tell people to look up the installation instructions of these things (e.g it's not our problem)

**Review by ValentinS4t1qbit — CHANGES_REQUESTED** [review]
A question regarding windows for test, and I think our README is quite bloated already: we need to be as smart and concise as possible.

Pretty sure I will make another PR myself to go over this README again (I'm picky) but let's just focus on your additions in this PR.

File: .github/workflows/run_psi4_test.yml (hunk 5 above)
Comment (JamesB-1qbit) [reply .github/workflows/run_psi4_test.yml:8]: I was mostly using it to test whether we need to direct people to use Windows Linux Subsystem. I can change it back. But the tests are now passing and it appears people can use Tangelo in Windows (native) with Psi4.

Comment (ValentinS4t1qbit) [reply .github/workflows/run_psi4_test.yml:8]: Great information. Let's stick to Ubuntu now and make the matter of extending things to Windows / Mac / whatever the scope of a subsequent PR, to do things in a smart way.

File: .github/workflows/run_psi4_test.yml [hunk 7]
```diff
@@ -29,42 +26,36 @@ jobs:
     - name: Install psi4
       shell: bash -el {0}
       run: |
-        conda create -n p4env psi4 -c psi4
+        conda install psi4 python=3.8 -c conda-forge/label/libint_dev -c conda-forge
```
Comment (ValentinS4t1qbit) [inline .github/workflows/run_psi4_test.yml:29]: also meh, python 3.8 hardcoded. Let's revert back to the general ubuntu image we had, and we'll try to be smart about windows and such later

File: README.rst (hunk 2 above)
Comment (JamesB-1qbit) [reply README.rst:134]: I think there should be a notebook to some other package but the link, to the test, I added  should be good enough for now.

File: README.rst (hunk 3 above)
Comment (JamesB-1qbit) [reply README.rst:130]: I changed to Classical Quantum Chemistry Packages. Perhaps there should be a hyphen or something.

File: .github/workflows/run_psi4_test.yml (hunk 7 above)
Comment (JamesB-1qbit) [reply .github/workflows/run_psi4_test.yml]: Psi4 only supports python 3.8 for Windows (native). Reverted

**Review by ValentinS4t1qbit — APPROVED** [review]
Cool beans ! I think at this point it's mostly about potential improvements or beautification but we have met the requirements of our release.

If we see anything about:

- testing / Docker
- using environment variables to let people choose a q.chem backend if several are installed or provide a path to it or something useful
- upgrading our documentation or examples
- bug

then it's a good time to talk about it quickly and sneak a PR in. If we don't see anything I think that's fine
